src/uct/tcp/tcp_iface.c

```diff
@@ -140,15 +159,18 @@ static ucs_status_t uct_tcp_iface_query(uct_iface_h tl_iface, uct_iface_attr_t *
                              UCT_IFACE_FLAG_AM_SHORT         |
                              UCT_IFACE_FLAG_AM_BCOPY         |
                              UCT_IFACE_FLAG_PENDING          |
-                             UCT_IFACE_FLAG_CB_SYNC          |
-                             UCT_IFACE_FLAG_ERRHANDLE_PEER_FAILURE;
+                             UCT_IFACE_FLAG_CB_SYNC;
     attr->cap.event_flags  = UCT_IFACE_FLAG_EVENT_SEND_COMP |
                              UCT_IFACE_FLAG_EVENT_RECV      |
                              UCT_IFACE_FLAG_EVENT_FD;
 
     attr->cap.am.max_short = am_buf_size;
     attr->cap.am.max_bcopy = am_buf_size;
 
+    if (uct_tcp_iface_is_keepalive_enabled(iface)) {
+        attr->cap.flags   |= UCT_IFACE_FLAG_ERRHANDLE_PEER_FAILURE;
```

Comment:
why?
we could detect peer failure when doing send and recv (if the peer crashed/closed)
should we set `UCT_IFACE_FLAG_EP_CHECK` based on TCP keepalive feature support?

Reply:
detection failure on receive depends from default kernel settings & may be disabled by default

Reply:
@hoopoepg, can you please elaborate? Why need to make err handling depending on KA?

Reply:
reverted